Mathlib/RingTheory/TensorProduct/MonoidAlgebra.lean

```diff
@@ -0,0 +1,82 @@
+/-
+Copyright (c) 2025 Michał Mrugała. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Michał Mrugała
+-/
+import Mathlib.Algebra.MonoidAlgebra.Basic
+import Mathlib.RingTheory.IsTensorProduct
+
+/-!
+# Monoid algebras are invariant under base change
+
+In this file we show that monoid algebras are stable under pushout.
+
+## TODO
+
+Additivise
+-/
+
+noncomputable section
+
+open TensorProduct
+
+namespace MonoidAlgebra
+variable {R M S A B : Type*} [CommSemiring R] [CommSemiring S] [Algebra R S] [CommMonoid M]
+variable [CommSemiring A] [Algebra R A] [CommSemiring B] [Algebra R B]
+
+variable (R A B) in
+/-- Tensoring `MonoidAlgebra R M` on the left by an `R`-algebra `A` is algebraically
+equivalent to `MonoidAlgebra A M`. -/
+noncomputable def tensorEquiv : A ⊗[R] MonoidAlgebra B M ≃ₐ[A] MonoidAlgebra (A ⊗[R] B) M := by
+  refine .ofAlgHom
+    (Algebra.TensorProduct.lift
+      ((IsScalarTower.toAlgHom A (A ⊗[R] B) _).comp Algebra.TensorProduct.includeLeft)
+      (mapRangeAlgHom _ Algebra.TensorProduct.includeRight)
+      (fun p n => .all _ _))
+    (MonoidAlgebra.liftNCAlgHom (Algebra.TensorProduct.map (.id _ _) singleOneAlgHom)
+        ((Algebra.TensorProduct.includeRight.toMonoidHom.comp (of B M))) (fun _ _ ↦ .all _ _)) ?_ ?_
+  · classical
+    apply AlgHom.toLinearMap_injective
+    ext
+    simp [liftNCAlgHom, liftNCRingHom, single_apply]
```

Comment:
Are we missing a `liftNCAlgHom_single` lemma here?

Reply:
Nope :)

Reply:
So what's the issue here?

Reply:
No issue, the lemmas are unnecessary. You're too quick at checking your inbox ;)